notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-1165_m1m3_bump_test_error_and_measured_forces.ipynb

```diff
@@ -0,0 +1,1060 @@
+    "            marker=\"o\",\n",
+    "            linestyle=\"None\",\n",
+    "            s=10,\n",
+    "            facecolor=\"none\",\n",
+    "            edgecolor=orientation_colors[orientation],\n",
+    "            alpha=0.9,\n",
+    "            label=f\"Orientation {orientation}\",\n",
+    "        )\n",
+    "    for combined_orientation in combined_orientations:\n",
+    "        ax.scatter(\n",
+    "            [],\n",
+    "            [],\n",
+    "            marker=\"o\",\n",
+    "            linestyle=\"None\",\n",
+    "            s=10,\n",
+    "            facecolor=\"none\",\n",
+    "            edgecolor=combined_orientations_colors[combined_orientation],\n",
+    "            alpha=0.5,\n",
+    "            label=f\"Orientation {combined_orientation}\",\n",
+    "        )\n",
+    "    ax.legend(loc=\"upper left\", bbox_to_anchor=(1, 1))\n",
+    "\n",
+    "    return"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4cb1a4c9",
+   "metadata": {},
+   "source": [
+    "## Histogram + Spatial distribution\n",
+    "\n",
+    "Here, we first process the log messages to extract the measured forces and return a dataframe with the measured forces, direction of the force measured, deviations, etc. Then, we use the helper function defined on sections above to create a bar plot of the failures. In order to see the spatial distribution of the actuators that failed the bump test, we also plot the spatial distribution of the failed actuators indicating the direction of the failed actuators."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2aac1bc0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Processing the bump logs\n",
+    "processed_bump_logs = process_bump_logs(bump_logs)\n",
+    "df = processed_bump_log.copy()\n",
+    "\n",
+    "df.tail()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "59dee03d-fe8f-4f5d-ac31-0683b29f6bf2",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2024-02-05T19:22:50.973679Z",
+     "iopub.status.busy": "2024-02-05T19:22:50.973040Z",
+     "iopub.status.idle": "2024-02-05T19:22:52.389238Z",
+     "shell.execute_reply": "2024-02-05T19:22:52.388689Z",
+     "shell.execute_reply.started": "2024-02-05T19:22:50.973660Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Create the figure and axes\n",
+    "fig, [ax0, ax1] = plt.subplots(1, 2, figsize=(13, 5))\n",
+    "\n",
+    "# Histogram aspect should be automatic\n",
+    "ax0.set_aspect(\"auto\", adjustable=\"box\")\n",
+    "\n",
+    "# Layout figure aspect needs to be rectangular\n",
+    "ax1.set_aspect(\"equal\", adjustable=\"box\")\n",
+    "\n",
+    "# Call the function to plot the actuators and flag the failed ones\n",
+    "plot_failure_histogram(df, ax0)\n",
+    "\n",
+    "# Call the function to plot the actuators and flag the failed ones\n",
+    "ActuatorsLayout(ax1, df, actuator_list=FATable)\n",
```

Comment:
I might miss but if there is any legend or text that what stars shapes are  (hardpoints).